### PR TITLE
Barking 6

### PR DESCRIPTION
**Pull request by thanosbnt**
### Description

Added Active people survey importer
### Checklist

- [ ] Created new test(s) (if applicable)
- [ ] Updated the README / documentation (if applicable)

### CODE REVIEW

File: src/main/java/uk/org/tombolo/importer/dft/ActivePeopleSurveyImporter.java [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+        int ignore = 0;
+        while (ignore++ < 7) {
+            rowIterator.next();
+        }
+
+        // Getting the indices of the columns containing the data.
+        int columnIndex = 0;
+        List<Integer> columnLoop = new ArrayList<>();
+        while ( columnIndex++ <= 24) {
+            if (!(columnIndex % 5 == 0 && columnIndex != 0)) {
```
Comment (lorenaqendro) [inline src/main/java/uk/org/tombolo/importer/dft/ActivePeopleSurveyImporter.java]: a negated `and` could be an `or` 🤓

File: src/test/java/uk/org/tombolo/importer/ons/OaImporterTest.java [hunk 2]
```diff
@@ -113,4 +121,17 @@ public void testImportLocalAuthorities() throws Exception {
...
+        importer.importDatasource("englandBoundaries", null, null, null);
+        SubjectType subjectType = SubjectTypeUtils.getSubjectTypeByProviderAndLabel(importer.getProvider().getLabel(), "englandBoundaries");
+        Subject englandboundaries = SubjectUtils.getSubjectByTypeAndLabel(subjectType,"E12000001");
+
+        assertEquals("North East", englandboundaries.getName());
+        assertEquals("englandBoundaries", englandboundaries.getSubjectType().getLabel());
+        assertEquals(-1.9039748587373244, englandboundaries.getShape().getCentroid().getX(), 0.1E-6);
+        assertEquals(55.019350338323875, englandboundaries.getShape().getCentroid().getY(), 0.1E-6);
+        assertEquals(9, importer.getSubjectCount());
+    }
```
Comment (lorenaqendro) [inline src/test/java/uk/org/tombolo/importer/ons/OaImporterTest.java]: add new line please

File: src/test/resources/datacache/TomboloData/uk.gov.ons/d9bcaf15-6f9b-381b-83a1-cf6505f53a11.json [hunk 3]
```diff
@@ -0,0 +1 @@
...8631227942,53.45951998549358],[-0.4717494668253328,53.474872899991965],[-0.4866300047912891,53.4804701659859],[-0.4883836593611129,53.50486022362468],[-0.40507452637143926,53.51761690248537],[-0.40814508052708404,53.532108866801394],[-0.4306647499876018,53.546337154659774],[-0.501229126936907,53.53746121569375],[-0.49092693046665214,53.545874921981195],[-0.4670183729790491,53.5489606915057],[-0.4290010461133578,53.574524713324664],[-0.4169629745636542,53.56321468811861],[-0.33593882116518226,53.55862659044057],[-0.300743141013308,53.595283629179406],[-0.3085552483220068,53.61432875698207],[-0.29211512923366634,53.61326743032027],[-0.24847604914788585,53.59366369079188]]]}},{"type":"Feature","properties":{"objectid":7,"rgn16cd":"E12000007","rgn16nm":"London","bng_e":517516,"bng_n":178392,"long":-0.30864,"lat":51.49227142,"st_areashape":1585474037.1241376,"st_lengthshape":270856.55508814706},"geometry":{"type":"Polygon","coordinates":[[[-0.011918676012088217,51.680877814680166],[-0.01225968853840528,51.64622869631537],[0.022719541656957977,51.641115492903985],[0.02181903530439697,51.62883278612958],[0.07283222796589423,51.60468738374122],[0.13818423542889827,51.62354539607562],[0.200
```
Comment (lorenaqendro) [inline src/test/resources/datacache/TomboloData/uk.gov.ons/d9bcaf15-6f9b-381b-83a1-cf6505f53a11.json]: add new line at the end

File: src/main/java/uk/org/tombolo/importer/dft/ActivePeopleSurveyImporter.java [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+    // Instantiating the datasource specifications.
+    @Override
+    public DatasourceSpec getDatasourceSpec(String datasourceIdString) throws Exception {
+        return DatasourceId.valueOf(datasourceIdString).datasourceSpec;
+
+    }
+
+    @Override
+    protected List<String> getOaDatasourceIds() {
+        return Arrays.asList(
```
[the remark this answers is not in the document]
Comment (thanosbnt) [reply src/main/java/uk/org/tombolo/importer/dft/ActivePeopleSurveyImporter.java]: A third option would be to just ignore the england boundaries geometries, as they can be derived by aggregation. I think we could just leave them in the database, like you said there are only 9 in total and might be usefull for comparison purposes...

File: src/main/java/uk/org/tombolo/importer/dft/ActivePeopleSurveyImporter.java (hunk 1 above)
Comment (thanosbnt) [reply src/main/java/uk/org/tombolo/importer/dft/ActivePeopleSurveyImporter.java]: yes but you have to flip the conditions as well  `columnIndex % 5 != 0 || columnIndex == 0` 🤓🤓

File: src/main/java/uk/org/tombolo/importer/dft/ActivePeopleSurveyImporter.java [hunk 5]
```diff
@@ -0,0 +1,203 @@
...
+        year =  "20" + year.substring(year.length()-2);
+        LocalDateTime timestamp = TimedValueUtils.parseTimestampString(year);
+
+        while (rowIterator.hasNext()) {
+            Row row = rowIterator.next();
+            Subject subject = SubjectUtils.getSubjectByTypeAndLabel(localauthority, String.valueOf(row.getCellByIndex(0).getDisplayText()).trim());
+            Subject subjecteb = SubjectUtils.getSubjectByTypeAndLabel(englandboundaries, String.valueOf(row.getCellByIndex(0).getDisplayText()).trim());
+
+            if (subject != null) {
+                for (int i = 0; i < columnLoop.size(); i++) {
```
[the remark this answers is not in the document]
Comment (thanosbnt) [reply src/main/java/uk/org/tombolo/importer/dft/ActivePeopleSurveyImporter.java]: 👍

File: src/main/java/uk/org/tombolo/importer/dft/ActivePeopleSurveyImporter.java [hunk 6]
```diff
@@ -0,0 +1,203 @@
...
+                    Attribute attribute = datasource.getTimedValueAttributes().get(i);
+
+                    timedValues.add(new TimedValue(
+                            subject,
+                            attribute,
+                            timestamp,
+                            record/100.));
+                }
+            }
+            else if (subjecteb != null) {
```
[the remark this answers is not in the document]
Comment (thanosbnt) [reply src/main/java/uk/org/tombolo/importer/dft/ActivePeopleSurveyImporter.java]: if both subjects return null then it is an invalid entry as far as i can tell...

File: src/main/java/uk/org/tombolo/importer/dft/ActivePeopleSurveyImporter.java [hunk 7]
```diff
@@ -0,0 +1,203 @@
...
+        if (datasourceId.equals("activePeopleWalk")) {
+            String[] elements = {
+                    "fractionWalk_1pm", "fractionWalk_1pw", "fractionWalk_3pw", "fractionWalk_5pw",
+                    "fractionWalkRecreation_1pm", "fractionWalkRecreation_1pw", "fractionWalkRecreation_3pw", "fractionWalkRecreation_5pw",
+                    "fractionWalkUtility_1pm", "fractionWalkUtility_1pw", "fractionWalkUtility_3pw", "fractionWalkUtility_5pw",
+                    "fractionWalk_less30mins", "fractionWalk_less60mins", "fractionWalk_less120mins", "fractionWalk_more120mins",
+                    "fractionWalkRecreation_less30mins", "fractionWalkRecreation_less60mins", "fractionWalkRecreation_less120mins", "fractionWalkRecreation_more120mins"
+            };
+            String[] description = {
+                    "% of adults that walk at least 1 x per month", "% of adults that walk at least 1 x per week", "% of adults that walk at least 3 x per week", "% of adults that walk at least 5 x per week",
```
[the remark this answers is not in the document]
Comment (thanosbnt) [reply src/main/java/uk/org/tombolo/importer/dft/ActivePeopleSurveyImporter.java]: Good catch

**Pull request merged**